packages/elasticsearch-asset-apis/src/elasticsearch-reader-api/WindowState.ts

```diff
@@ -36,8 +36,8 @@ export class WindowState {
 
         if (allDone) {
             for (const key of Object.keys(this._windowState)) {
-                this._windowState[key].canRestart = true;
-                this._windowState[key].hasCalled = false;
+                this._windowState[Number(key)].canRestart = true;
+                this._windowState[Number(key)].hasCalled = false;
```

Comment:
why does this have to happen, in the type above it already marked the keys as numbers, why does this have to happen again?